src/python/modules/ensembl/compara/citest/tabletest.py

```diff
@@ -0,0 +1,245 @@
+                self.test_results.append(result)
+                i=i+1
+
+
+    def _get_number_row_by_group(self,obj_connection_engine,str_table_name,str_group_by):
+        lst_result=[]
+        str_request="select count(*),"+str_group_by+" from "+str_table_name+" group by "+str_group_by+";"
+        obj_request=text(str_request)
+        obj_connection=obj_connection_engine.connect()
+        obj_result=obj_connection.execute(obj_request)
+        for tpl_res in obj_result:
+            lst_result.append(tpl_res)
+        obj_connection.close()
+        return sorted(lst_result,key=lambda tuple: tuple[1])# return a list sorted on the group_by value
+
+
+    def _test_table_by_column(self, str_table_name, lst_columns):
+
+        # transfornm the json based column in sinmple stroing list.
+        lst_col =[]
+        for column in lst_columns:
+            lst_col.append(column["name"])
+
+        dic_column_ref=self._get_column_values(self.ref_connection_engine, str_table_name, lst_col)
+        dic_column_tar=self._get_column_values(self.target_connection_engine, str_table_name, lst_col)
+
+        for str_column, lst_fields_ref in dic_column_ref.items():
+            lst_fields_tar=dic_column_tar[str_column]
+            i=0
+            result={}
+            result["succes"]=1
+            str_val_ref=""
+            str_val_tar=""
+            while i < len(lst_fields_ref):
+                if lst_fields_ref[i] != lst_fields_tar[i]:
+                    result["succes"]=0
+                    str_val_ref=lst_fields_ref[i]
+                    str_val_tar=lst_fields_tar[i]
+                    break
+                i=i+1
+            str_info="col:"+str_column+"|val_ref:"+str(str_val_ref)+"|val_tar:"+str(str_val_tar)
+            result["info"]=str_info
+            result["table"]=str_table_name
+            result["type"]="column"
+            self.test_results.append(result)
+
+
+    def _get_column_values(self, obj_connection_engine, str_table, lst_columns):
+        # init dictionary
+        dic_column={}
+        str_request=""
+        bool_tag_all=False
+
+        #look if there is a "*" in the list of fields
+        for str_column in lst_columns:
+            if str_column=="*":
+                bool_tag_all=True
+
+        #if there is a star selct all, toherise select the selected tables
+        if bool_tag_all:
+            str_request="select * from "+str_table+";"
+        elif len(lst_columns)==1:
+            str_request="select "+lst_columns[0]+" from "+str_table+";"
+        else:
+            str_request="select "+",".join(lst_columns)+" from "+str_table+";"
+
+        obj_request=text(str_request)
+        obj_connection=obj_connection_engine.connect()
+        obj_result=obj_connection.execute(obj_request)
+
+
+        for str_col in obj_result.keys():
+            # if there is a "*" in the lst column then all other columns are
+            # defined to be excluded so we exclud them
```

Comment:
I don't really like this method of including/excluding columns. It makes the JSON unclear. JSON allows for such flexibility, I don't know why we wouldn't just add an `exclude_columns` field (or similar) for this purpose.